_posts/en/newsletters/2020-07-01-newsletter.md

```diff
@@ -0,0 +1,371 @@
+slug: 2020-07-01-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter summarizes a discussion about mining incentives
+related to HTLCs and links to an announcement about a proposed service to
+store and relay presigned transactions.  Also included are our regular
+sections with recently transcribed talks and conversations, new releases
+and release candidates, and notable changes to popular Bitcoin
+infrastructure projects.
+
+## Action items
+
+*None this week.*
+
+## News
+
+{% comment %}<!-- FIXME Reviewers: the following item links to two of my
+posts; please be extra critical of those links and their related
+content.  I'll respect any decisions you make regarding them -->{% endcomment %}
+
+- **Discussion of HTLC mining incentives:** Itay Tsabary, Matan
+  Yechieli, Ittay Eyal [posted][tye post] to the Bitcoin-Dev mailing
+  list with a paper they've written that claims rational miners should
+  want to run a modified Bitcoin node that allows a user to bribe it
+  with a transaction that's timelocked and can't be confirmed until some
+  point in the future.  As a result of this bribe, the miner won't
+  confirm any transaction that could be mined immediately but which
+  conflicts with the bribe (as long as the bribe pays a sufficiently
+  higher feerate than any transactions it blocks).
+
+    If this theory is accurate, it affects Hash TimeLock Contracts
+    (HTLCs) which can be settled immediately by one party (Alice) using
+    a preimage, or which can be settled later by a second party (Bob)
+    after a timeout.  According to the authors, if Bob sees Alice use
+    the preimage to spend the HTLC, Bob should send his conflicting
+    timeout settlement transaction to miners with a sufficiently higher
+    feerate than Alice's preimage settlement---even though miners can't
+    immediately include Bob's transaction---bribing them to ignore
+    Alice's transaction in favor of waiting to confirm his alternative
+    transaction.  This allows Bob to steal the money Alice should
+    receive.  HTLCs are currently used in LN, atomic swaps, and several
+    other contract protocols.
+
+    The authors of the paper propose a solution they call Mutually
+    Assured Destruction HTLCs (MAD-HTLCs).  These require Bob to provide a
+    hashlocked bond that makes him reveal his own preimage when sending
+    his timeout settlement.  If both Alice and Bob reveal their
+    respective preimages, miners will be able to claim both the
+    payment/refund amount and the bond collateral---destroying the
+    incentive for Bob to attempt stealing by bribing miners.
+
+    Downsides to this approach are that it requires Bob to use more
+    collateral---raising the cost of using HTLCs---and it uses more
+    block chain space when MAD-HTLCs are settled onchain compared to
+    traditional HTLCs---also raising costs.  It was also
+    [claimed][harding mad miner] in the mailing list discussion that
```

Comment:
I understand why the author here is not named here and at the end of this paragraph, but it might be valuable to readers to add "by this author" just as other valued participants would be named. Feel free to ignore; I don't want to waste your time :)

Reply:
You're quite likely right that it would be valuable to readers.  However, you can be sure that as long as I have a say in it, anywhere there's a spotlight, it'll be pointed as far away as possible from me.  :-)